### PR TITLE
Check trap code in test suite assertions

### PR DESCRIPTION
**Pull request by Mrowqa**
Depends on https://github.com/CraneStation/wasmtime/pull/357.

### CODE REVIEW

**Comment by Mrowqa** [comment]
Tests: traps, i32, i64, int_exprs fails on `0x80000000(00000000) / -1` which is expected to trap on integer oveflow, but cranelift currently says it's division by zero.

**Comment by Mrowqa** [comment]
The rest of the tests (call_indirect, elem, imports, linking) can be solved by replacing `"bad signature"` trap message with `"bad signature (indirect call type mismatch | uninitialized element 7)".to_string()`. Every one of these tests has its own name for this trap. Moreover, I have noticed that some assert messages are substrings of actual expected messages (i.e. `undefined` instead of `undefined element`). So, testing if trap message contains expected message doesn't seem ideal.

File: wasmtime-runtime/src/traphandlers.rs [hunk 1]
```diff
@@ -84,13 +84,30 @@ fn trap_message() -> String {
...
 }
 
+fn trap_code_to_expected_string(trap_code: ir::TrapCode) -> String {
+    use ir::TrapCode::*;
+    match trap_code {
+        StackOverflow => "call stack exhausted".to_string(),
+        HeapOutOfBounds => "out of bounds memory access".to_string(),
+        TableOutOfBounds => "undefined element".to_string(),
+        OutOfBounds => "out of bounds".to_string(), // TODO: currently not covered by the test suite
+        IndirectCallToNull => "indirect call to null".to_string(), // TODO: currently not covered by the test suite
```
Comment (Mrowqa) [inline wasmtime-runtime/src/traphandlers.rs]: undefined element

File: wasmtime-runtime/src/traphandlers.rs [hunk 2]
```diff
@@ -84,13 +84,30 @@ fn trap_message() -> String {
...
+        BadSignature => "bad signature".to_string(), // TODO: various tests have different names for this trap
```
Comment (Mrowqa) [inline wasmtime-runtime/src/traphandlers.rs]: `"indirect call type mismatch"`

**Comment by Mrowqa** [comment]
TODO:
- [ ] set `avoid_div_traps` flag in the builder.

**Comment by sunfishcode** [comment]
The CI failure here is due to stable rust's rustfmt changing, and isn't related to this PR.

**Pull request merged**